src/PaymentGateways/PayPalCommerce/Repositories/PayPalOrder.php

```diff
@@ -18,180 +18,206 @@
-			logError(
-				'Create PayPal Commerce payment refund failure',
-				sprintf(
-					'<strong>Response</strong><pre>%1$s</pre>',
-					print_r( json_decode( $exception->getMessage(), true ), true )
-				)
-			);
-
-			throw $exception;
-		}
-	}
-
-	/**
-	 * Validate argument given to create PayPal order.
-	 *
-	 * @since 2.9.0
-	 *
-	 * @param  array  $array
-	 *
-	 * @throws InvalidArgumentException
-	 */
-	private function validateCreateOrderArguments( $array ) {
-		$required = [ 'formId', 'donationAmount', 'payer' ];
-		$array    = array_filter( $array ); // Remove empty values.
-
-		if ( array_diff( $required, array_keys( $array ) ) ) {
-			throw new InvalidArgumentException(
-				__(
-					'To create a paypal order, please provide formId, donationAmount and payer',
-					'give'
-				)
-			);
-		}
-	}
+class PayPalOrder
+{
+    /**
+     * @since 2.9.0
+     *
+     * @var PayPalClient
+     */
+    private $paypalClient;
+
+    /**
+     * @since 2.9.0
+     *
+     * @var MerchantDetail
+     */
+    private $merchantDetails;
+
+    /**
+     * PayPalOrder constructor.
+     *
+     * @since 2.9.0
+     *
+     * @param MerchantDetail $merchantDetails
+     *
+     * @param PayPalClient $paypalClient
+     */
+    public function __construct(PayPalClient $paypalClient, MerchantDetail $merchantDetails)
+    {
+        $this->paypalClient    = $paypalClient;
+        $this->merchantDetails = $merchantDetails;
+    }
+
+    /**
+     * Approve order.
+     *
+     * @since 2.9.0
+     *
+     * @param string $orderId
+     *
+     * @return string
+     * @throws Exception
+     */
+    public function approveOrder($orderId)
+    {
+        $request = new OrdersCaptureRequest($orderId);
+
+        try {
+            return $this->paypalClient->getHttpClient()->execute($request)->result;
+        } catch (Exception $ex) {
+            logError(
+                'Capture PayPal Commerce payment failure',
+                sprintf(
+                    '<strong>Response</strong><pre>%1$s</pre>',
+                    print_r(json_decode($ex->getMessage(), true), true)
+                )
+            );
+
+            throw $ex;
+        }
+    }
+
+    /**
+     * Create order.
+     *
+     * @since 2.9.0
+     *
+     * @param array $array
+     *
+     * @return string
+     * @throws Exception
+     * @see https://developer.paypal.com/docs/api/orders/v2
+     *
+     */
+    public function createOrder($array)
+    {
+        $this->validateCreateOrderArguments($array);
+
+        $request          = new OrdersCreateRequest();
+        $formId           = (int)$array['formId'];
+        $donationCurrency = give_get_currency($formId);
+        $donationAmount   = give_maybe_sanitize_amount(
+            $array['donationAmount'],
+            ['currency' => give_get_currency($formId)]
+        );
+        $request->payPalPartnerAttributionId(give('PAYPAL_COMMERCE_ATTRIBUTION_ID'));
+        $request->body = [
```

Comment:
@JasonTheAdams  code automatically formated in PHPStorm. i made changes on in PayPal payment request body